hack/e2e.sh

```diff
@@ -292,6 +292,7 @@ if [ "$PROVIDER" == "gke" ]; then
         --test exec
         -v 1
         --cluster-name "$CLUSTER"
+        --network "$CLUSTER"
```

Comment:
same as before ([old gke job](https://github.com/kubernetes/test-infra/blob/4afd9ba744d65d6a3ff028118a4adb1a00026e7d/scenarios/kubernetes_e2e.py#L534-L535), we use the cluster name as the network name